src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -72,9 +72,7 @@ protected override MethodBuilder DefineMethodCore(string name, MethodAttributes
         protected override void SetCustomAttributeCore(CustomAttributeBuilder customBuilder) => throw new NotImplementedException();
 
         [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2074:DynamicallyAccessedMembers",
-            Justification = "No need to propogate the attriubte for called method")]
-        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
-            Justification = "Do not expect System.Object type will be trimmed from core assembly")]
+            Justification = "No need to propogate the attribute to the called method")]
```

Comment:
Just curious - why do we still need this UnconditionalSuppressMessage ?

Reply:
The parameter `parent` has [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] attribute, therefore was `_typeParent` had to have that attribute. Now on assignment: 
```cs
_typeParent = _module.GetTypeFromCoreAssembly(CoreTypeId.Object);
``` 
its warning `Value stored in field does not satisfy 'DynamicallyAccessedMembersAttribute' requirements. The return value of the source method does not have matching annotations.` 

I believe we don't want to add the attribute to `GetTypeFromCoreAssembly(CoreTypeId typeId)`

Reply:
> I believe we don't want to add the attribute to GetTypeFromCoreAssembly(CoreTypeId typeId

If we don't how can we guarantee that trimming will leave the type as-is. The requirement for `All` means trimming need to preserve the type as a whole. The code above (`s_coreTypes`) only guarantees that we preserve the existence of all the relevant types, but it does not guarantee that members on those types will be preserved. If the code here requires that, then we need to add additional dependencies to make that happen.

Reply:
> The parameter parent has [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] attribute, therefore was _typeParent had to have that attribute.

Why does `_typeParent` field need to have the attribute? It should not need it and it should be ok to drop annotations on assignments.

Reply:
> Why does _typeParent field need to have the attribute? It should not need it and it should be ok to drop annotations on assignments.

I think it needs at the very least default constructors preserved - since the type builder implementation will look for that on its own. But I vaguely remember that there were also cases with overrides where it was needed, but I could be wrong.
https://github.com/dotnet/runtime/blob/e579ccb60acb044453f2cf1907482b165658488e/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeTypeBuilder.cs#L1393

It is unfortunately a place where the implementation behavior leaks into the interface (abstract base) - but I think it's the only thing necessary to make `RuntimeTypeBuilder` work with trimming.

Reply:
> Adding `[return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]` attribute to the Type GetTypeFromCoreAssembly(CoreTypeId typeId)

This alone won't preserve the constructor.

We only really need to worry about preserving Object constructor here. Preserving other constructors is a problem for user code. It should be easier to special case the Object constructor rather than try to make GetTypeFromCoreAssembly to do it.

For this PR, I think it is to just add TODO next to UnconditionalSuppressMessage

Reply:
> For this PR, I think it is to just add TODO next to UnconditionalSuppressMessage

Done